File: Android/doraemonkit-no-op/src/main/java/com/didichuxing/doraemonkit/ui/base/DokitIntent.java
package com.didichuxing.doraemonkit.ui.base;

import android.app.Activity;
import android.os.Bundle;

/**
 * Created by jintai on 2019/9/16.
 * dokitView intent
 */

public class DokitIntent {
    public static final int MODE_NORMAL = 0;

    public static final int MODE_SINGLE_INSTANCE = 1;

    public Class<? extends Object> targetClass;

    public Bundle bundle;
    /**
     * 赋值为类名 不接受对外赋值
     */
    private String tag;

    public String getTag() {
        return tag;
    }

    public Activity activity;

    public int mode = MODE_NORMAL;


    public DokitIntent(Class<? extends Object> targetClass) {

    }

}

File: Android/dokit-framework/src/main/java/com/didichuxing/doraemonkit/kit/core/DoKitIntent.kt
package com.didichuxing.doraemonkit.kit.core

import android.app.Activity
import android.os.Bundle
import com.didichuxing.doraemonkit.extension.tagName
import com.didichuxing.doraemonkit.util.ActivityUtils
import kotlin.reflect.KClass

/**
 * Created by jintai on 2019/9/16.
 * dokitView intent
 */
data class DoKitIntent(
    var targetClass: Class<out AbsDoKitView>,
    var activity: Activity = ActivityUtils.getTopActivity(),
    var bundle: Bundle? = null,
    var tag: String = targetClass.tagName,
    var mode: DoKitViewLaunchMode = DoKitViewLaunchMode.SINGLE_INSTANCE
)


enum class DoKitViewLaunchMode {
    SINGLE_INSTANCE,
    /**
     * 倒计时
     */
    COUNTDOWN
}
